rpc/types/src/status.rs

```diff
@@ -0,0 +1,182 @@
+/// ```
+#[derive(
+    Copy, Clone, Debug, Default, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize,
+)]
+pub enum Status {
+    // FIXME:
+    // `#[serde(rename = "")]` only takes raw string literals?
+    // We have to re-type the constants here...
+    /// Successful RPC response, everything is OK; [`CORE_RPC_STATUS_OK`].
+    #[serde(rename = "OK")]
+    #[default]
+    Ok,
+
+    /// The daemon is busy, try later; [`CORE_RPC_STATUS_BUSY`].
+    #[serde(rename = "BUSY")]
+    Busy,
+
+    /// The daemon is not mining; [`CORE_RPC_STATUS_NOT_MINING`].
+    #[serde(rename = "NOT MINING")]
+    NotMining,
+
+    /// Payment is required for RPC; [`CORE_RPC_STATUS_PAYMENT_REQUIRED`].
+    #[serde(rename = "PAYMENT REQUIRED")]
+    PaymentRequired,
+
+    /// Some unknown other string; [`CORE_RPC_STATUS_UNKNOWN`].
+    ///
+    /// This exists to act as a catch-all if `monerod` adds
+    /// a string and a Cuprate node hasn't updated yet.
+    ///
+    /// The reason this isn't `Unknown(String)` is because that
+    /// disallows [`Status`] to be [`Copy`], and thus other types
+    /// that contain it.
+    #[serde(other)]
+    #[serde(rename = "UNKNOWN")]
+    Unknown,
+}
+
+impl From<String> for Status {
+    fn from(s: String) -> Self {
+        match s.as_str() {
+            CORE_RPC_STATUS_OK => Self::Ok,
+            CORE_RPC_STATUS_BUSY => Self::Busy,
+            CORE_RPC_STATUS_NOT_MINING => Self::NotMining,
+            CORE_RPC_STATUS_PAYMENT_REQUIRED => Self::PaymentRequired,
+            _ => Self::Unknown,
+        }
+    }
+}
+
+impl AsRef<str> for Status {
+    fn as_ref(&self) -> &str {
+        match self {
+            Self::Ok => CORE_RPC_STATUS_OK,
+            Self::Busy => CORE_RPC_STATUS_BUSY,
+            Self::NotMining => CORE_RPC_STATUS_NOT_MINING,
+            Self::PaymentRequired => CORE_RPC_STATUS_PAYMENT_REQUIRED,
+            Self::Unknown => CORE_RPC_STATUS_UNKNOWN,
+        }
+    }
+}
+
+impl Display for Status {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str(self.as_ref())
+    }
+}
+
+// [`Status`] is essentially a [`String`] when it comes to
+// (de)serialization, except when writing we usually have
+// access to a `&'static str` and don't need to allocate.
+//
+// See below for more impl info:
+// <https://github.com/Cuprate/cuprate/blob/bef2a2cbd4e1194991751d1fbc96603cba8c7a51/net/epee-encoding/src/value.rs#L366-L392>.
+impl EpeeValue for Status {
+    const MARKER: Marker = <String as EpeeValue>::MARKER;
+
+    fn read<B: Buf>(r: &mut B, marker: &Marker) -> cuprate_epee_encoding::Result<Self> {
+        let string = <String as EpeeValue>::read(r, marker)?;
+        Ok(Self::from(string))
+    }
+
+    fn should_write(&self) -> bool {
+        true
+    }
+
+    fn epee_default_value() -> Option<Self> {
+        // TODO: what is the default here?
+        Some(Self::default())
+    }
```

Comment:
Not sure what to put here. An empty string?

Reply:
I would put this as `Unknown` as that's what an empty string should deserialize to